test/rules-google-enhanced-ecommerce-fullstory.spec.ts

```diff
@@ -335,6 +335,17 @@ describe('Ruleset: Google Analytics Enhanced Ecommerce to FullStory', () => {
         expectEqual(eventName, 'refund');
         expectEqual(payload.id, 'T12345');
       });
+
+      it('gracefully handles a "clear" event', async () => {
```

Comment:
I would've expected this test to fail against the latest DLO production version, but it doesn't:

```
$ npm i && npm run test:browser:bootstrap
$ PLAYWRIGHT_DLO_SCRIPT_SRC=https://edge.fullstory.com/datalayer/v2/latest.js npm run test:browser -- --grep Google
```

Is there something else we should be asserting against? What's the current behavior in production without these changes? I'm happy to help with any test changes here since I recently introduced test environments and test harnesses to ruleset tests.

Reply:
Ah, yes.  The undesirable side effect is that the `FullStoryAppender` creates FS.events when the `ERROR` level is applied.  This is noisy and triggers rate limiting.  (Updated the original PR description with these details.)

So I suppose that a better test is that the data layer event is undefined AND no log event was created as a result.  In other parts of the code, we do `expectNoCalls(globalMock.console, 'log');`.  Is that still valid for the test harness?  Or does the test harness allow for a log sink that is scoped to each test case?  (The latter seems needed.)

Reply:
@ndonze-fs Sure, push it and I'll review.

Reply:
@van-fs As I'm working through this change locally, I'm running into an issue with the GA Event Measurement tests logging an operator error against the latest revision in this PR. You should be able to see this error logged in the test output locally by either running all Google Analytics tests:

```
$ npm test -- --grep 'Google Analytics'
```

...or running just the Event Measurement tests:

```
$ npm test -- --grep 'Google Analytics Event Measurement'
```

Output:

> Operator Error {"operator":"query","path":"dataLayer.push","reason":"Cannot convert undefined or null to object"}

Any ideas? This error is causing test failures when I start checking for errors and assert there are none.

Reply:
I see it now.  With the original problem is addressed, the `isEmptyData` function will throw because it's trying to look at keys for a null value.  Investigating the behavior.  This code was desired to prevent "empty" data from being sent to FullStory IIRC.